src/main/java/core/basesyntax/strategy/handlers/BalanceDataHandler.java

```diff
@@ -0,0 +1,43 @@
+package core.basesyntax.strategy.handlers;
+
+import core.basesyntax.exceptions.FruitsNameException;
+import core.basesyntax.storage.Storage;
+import java.util.Objects;
+
+public class BalanceDataHandler implements DataHandler {
+
+    @Override
+    public void processData(String fruit, int quantity) {
+        validateBalanceDate(fruit,quantity);
+        Storage.addFruits(fruit, quantity);
+    }
+
+    private void validateBalanceDate(String fruit, int quantity) {
+        if (fruit == null || fruit.isEmpty()) {
+            throw new FruitsNameException("Wrong fruit name: "
+                    + fruit);
+        }
+        if (quantity < 0) {
+            throw new RuntimeException("Wrong fruit quantity for "
+                    + fruit
+                    + ", quantity: "
+                    + quantity);
+        }
+    }
+
+    @Override
+    public boolean equals(Object object) {
```

Comment:
Do you need it here? (same for other handlers)

Reply:
It was a student's advice. I asked about it in the questions and still did not understand whether I should do it. Will rework it

Reply:
Actually, you don't need it here. The hashmap works by the key's hashcode and equals, not value 
Delete it everywhere